src/pczt/signer.rs

```diff
@@ -0,0 +1,38 @@
+use rand::{CryptoRng, RngCore};
+
+use crate::keys::SpendAuthorizingKey;
+
+impl super::Spend {
+    /// Signs the Sapling spend with the given spend authorizing key.
+    ///
+    /// It is the caller's responsibility to perform any semantic validity checks on the
+    /// PCZT (for example, comfirming that the change amounts are correct) before calling
+    /// this method.
+    pub fn sign<R: RngCore + CryptoRng>(
+        &mut self,
+        sighash: [u8; 32],
+        ask: &SpendAuthorizingKey,
+        rng: R,
+    ) -> Result<(), SignerError> {
+        let alpha = self.alpha.ok_or(SignerError::MissingSpendAuthRandomizer)?;
+
+        let rsk = ask.randomize(&alpha);
+        let rk = redjubjub::VerificationKey::from(&rsk);
+
+        if self.rk == rk {
+            self.spend_auth_sig = Some(rsk.sign(rng, &sighash));
```

Comment:
I went down a rabbit hole here verifying that https://github.com/zcash/librustzcash/issues/179 is fixed. [It is.](https://github.com/zcash/librustzcash/issues/179#issuecomment-2537459745)